packages/cra-template-typescript/template/src/service-worker.ts

```diff
@@ -0,0 +1,50 @@
+// This service worker can be customized!
+// See https://developers.google.com/web/tools/workbox/modules
+// for the list of available Workbox modules, or add any other
+// code you'd like.
+// You can also remove this file if you'd prefer not to use a
+// service worker, and the Workbox build step will be skipped.
+
+declare const self: ServiceWorkerGlobalScope;
+
+import {clientsClaim} from 'workbox-core';
+import {precacheAndRoute, createHandlerBoundToURL} from 'workbox-precaching';
+import {registerRoute} from 'workbox-routing';
+
+clientsClaim();
+
+// Precache all of the assets generated by your build process.
+// Their URLs are injected into the manifest variable below.
+// eslint-disable-next-line no-restricted-globals
+precacheAndRoute(self.__WB_MANIFEST); 
+
+// Set up App Shell-style routing, so that all navigation requests
+// are fulfilled with your index.html shell. Learn more at
+// https://developers.google.com/web/fundamentals/architecture/app-shell
+const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$');
+registerRoute(
+  // Return false to exempt requests from being fulfilled by index.html.
+  ({request, url}: {request: Request, url: URL}) => {
```

Comment:
This default logic is equivalent to what was used in the older `c-r-a` with the navigation route configuration, but is obviously more flexible, as it could be customized. The goal was to ship something equivalent by default.